modules/friendlytag.js

```diff
@@ -1074,7 +1074,24 @@ Twinkle.tag.callbacks = {
 		$.each(tags, addTag);
 
 		if( Twinkle.tag.mode === 'redirect' ) {
-			pageText += tagText;
+			// Check for all Rcat shell redirects (from #433)
+			if (pageText.match(/{{(?:redr|this is a redirect|r(?:edirect)?(?:.?cat.*)?[ _]?sh)/i)) {
+				// Regex courtesy [[User:Kephir/gadgets/sagittarius.js]] at [[Special:PermaLink/831402893]]
+				var oldTags = pageText.match(/(\s*{{[A-Za-z ]+\|)((?:[^|{}]*|{{[^|}]*}})+)(}})\s*/i);
+				pageText = pageText.replace(oldTags[0], oldTags[1]+tagText+oldTags[2]+oldTags[3]);
+			} else {
+				// Fold any pre-existing Rcats into taglist and under Rcatshell
+				var pageTags = pageText.match(/\n{{R(?:edirect)? .*?}}/img);
+				var oldPageTags ='';
+				if (pageTags) {
+					$.each(pageTags, function(pageIndex, pageTag) {
```

Comment:
Twinkle is littered with `$.each` (which we should fix), but from here on out I would use the native JavaScript [`Array.prototype.forEach`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach). So something like `pageTags.forEach(function (pageTag) {`. This should work on IE 9+. MediaWiki itself no longer has official JavaScript support for anything below IE 11 anyway.

Reply:
I'll take care of this one, you do the rest, deal?

On a more serious note, might it be worth putting together some sort of contributing.md guide?

Reply:
Yeah we'll remove all the other $.each's in a separate PR. Not urgent.

I agree a contributing guide is needed. Want to open an issue for that? We should rope in all maintainers.